### PR TITLE
Add return element for waitFor, add specific device when forward tcp on android device

### PR DESCRIPTION
**Pull request by baovu2512**

### CODE REVIEW

File: driver/lib/sessions/android.ts [hunk 1]
```diff
@@ -22,23 +22,26 @@ const setupNewAndroidDriver = async (caps) => {
...
     connectSocket(await observatoryWsUri, caps.retryBackoffTime, caps.maxRetryCount),
   ]);
 
 };
 
-export const getObservatoryWsUri = async (proxydriver) => {
+export const getObservatoryWsUri = async (proxydriver , caps) => {
   const urlObject = processLogToGetobservatory(proxydriver.adb.logcat.logs);
+  
+  let {udid, emPort} = await androidHelpers.getDeviceInfoFromCaps(caps);
```
Comment (shibupanda) [inline driver/lib/sessions/android.ts:36]: what emPort is? I have not seen the usage of it anywhere.

**Review by shibupanda — COMMENTED** [review]
@baovu2512 thanks for collaboration. 
Could you please add some tests for newly added command.

**Comment by baovu2512** [comment]
Hi @shibupanda , I'm new with these language. can you help me to review above changes ?

**Comment by shibupanda** [comment]
Hi @baovu2512 could you please add a sample steps in script for the feature you have added so it user can understand the usage.

**Comment by baovu2512** [comment]
Hola @shibupanda , I added sample in java.

**Review by shibupanda — APPROVED** [review]

**Pull request merged**